### PR TITLE
EZP-21998: Implement @ParamConverter in controllers

### PR DESCRIPTION
**Pull request by yannickroger**
Link: https://jira.ez.no/browse/EZP-21998

## Description
Modification introduced by adding a Param Converter for Location and Content in the kernel: https://github.com/ezsystems/ezpublish-kernel/pull/1128/

## Test
Manual tests. And see if BDD works (thanks to #133) ;)

## TODO
- [x] Some more sanity test
- [ ] Remove commit pointing on dev branch

### CODE REVIEW

File: Helper/PlaceHelper.php [hunk 1]
```diff
@@ -66,16 +67,14 @@ public function __construct(
...
      *
-     * @param int|string $locationId id of a place_list
+     * @param \eZ\Publish\API\Repository\Values\Content\Location $location of a place_list
      * @param string|string[] $contentTypes to be retrieved
      * @param string|string[] $languages to be retrieved
      *
      * @return \eZ\Publish\API\Repository\Values\Content\Content[]
      */
-    public function getPlaceList( $locationId, $contentTypes, $languages = array() )
+    public function getPlaceList( $location, $contentTypes, $languages = array() )
```
Comment (andrerom) [inline Helper/PlaceHelper.php]: Missing type hint?

Comment (yannickroger) [inline Helper/PlaceHelper.php]: Yes ! Major update coming up.

**Comment by andrerom** [comment]
+1 :)

**Comment by yannickroger** [comment]
Ready for review

**Comment by bdunogier** [comment]
Why aren't the tests failing ?

**Comment by bdunogier** [comment]
+1

**Comment by yannickroger** [comment]
@bdunogier as you told me by chat: these modified part are not covered

I will try to create a first BDD test and see how much I can do in a timeboxed period.

**Comment by bdunogier** [comment]
To be frank, Travis should have failed.

We have changed a couple controllers to only accept an object as their argument. Without the ezpublish-kernel branch, they will NOT receive an object but an ID.

Blog posts list and feedback form are not covered, and are not expected to fail in behat. But the breadcrumb is on every page, and it should have failed. I'm getting this on a master install + this branch on DemoBundle, as expected:

```
An exception has been thrown during the rendering of a template ("Controller "EzSystems\DemoBundle\Controller\DemoController::viewBreadcrumbAction()" requires that you provide a value for the "$location" argument (because there is no default value or because there is a non optional argument after this one).") in eZDemoBundle::pagelayout.html.twig at line 56.
```

**Comment by bdunogier** [comment]
Okay, it doesn't fail tests because they're executed in prod (and that's good). The breadcrumb fails, but it doesn't prevent the pages frm being rendered (good).

But the tests don't fail because the breadcrumb isn't tested by any scenario (bad).

**Comment by bdunogier** [comment]
Now it fails as expected. I'm just not sure why it started failing on premium content... :-)

**Comment by yannickroger** [comment]
@bdunogier 
I moved the BDD test into a dedicated PR #133  So we can make sure they run on master. Once merged, I'll rebase this one and we can see if it still works.

**Comment by yannickroger** [comment]
Rebased with the BDD PR #133 merged and it passes. So this one should be ready once the kernel PR is merged. ping @bdunogier

**Comment by andrerom** [comment]
+1

**Comment by bdunogier** [comment]
+1 from me. Let's put this behind us and move on :-)

**Comment by andrerom** [comment]
@yannickroger do you remove the temporary thing and either merge or if travis has a vote here let it run true once?

**Comment by yannickroger** [comment]
Temporary stuff removed (commit here and branch on community), waiting for travis to merge.

**Comment by yannickroger** [comment]
still some errors on travis that look kernel related. ping @andrerom @bdunogier

**Pull request merged**